### ISSUE TITLE
Ignore single declaration block in force-*-nesting

### ISSUE THREAD

**Issue by ncoden** [sasstools/sass-lint#950]
Like, #827, I think the nesting forcing rules are too restrictive. When I have to apply properties to only one particular case of an element (pseudo-class / pseudo-element / chained class...), I do not expect to have to nest the pseudo-class/pseudo-element/class inside the element.

**Should throw an error:** Factorizable multiple declaration blocks
```scss
.element:pseudo-class { ... }
.element:other-pseudo-class { ... }
```
```scss
.element.chained-class { ... }
.element:pseudo-class { ... }
.element::pseudo-element { ... }
```

**Should not throw an error:** Single declaration block
```scss
.element:pseudo-class { ... }
```
```scss
.element.chained-class { ... }
.other-element.chained-class { ... }
```

This behavior could be enabled/disabled with an option like:
```yml
force-element-nesting:
  - 1
  - ignore-single-declaration: true
```

@DanPurdy What do you think ?

**Comment by DanPurdy** [comment]
I think it's definitely valid, the force nesting rules are far too strict right now and definitely need tuning.

**Comment by ncoden** [comment]
@DanPurdy Hi from 2 years later 👋 
What's the status of this ?
